File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.cpp
class Solution {
public:
   int canBeTypedWords(string text, string brokenLetters) {
    unordered_set<int> broken(begin(brokenLetters), end(brokenLetters));
    int res = 0;
    for (int i = 0, cnt = 0; i <= text.size(); ++i)
        if (i == text.size() || text[i] == ' ') {
            res += cnt == 0;
            cnt = 0;
        }
        else if(cnt == 0)
            cnt += broken.count(text[i]);
    return res;
    }
};

File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.java
class Solution {
    public int canBeTypedWords(String text, String broken) {
          HashSet<Character> set = new HashSet<>();
        for(char ch : broken.toCharArray()) set.add(ch);
        
        boolean flag = false;
        int res = 0;
        for(char ch : text.toCharArray()) {
            if(ch == ' ') {
                if(!flag) res++;
                flag = false;
            } else {
                if(set.contains(ch)) flag = true;
            }
        }
        
        return res += flag == false ? 1 : 0;
    }
}